core/src/main/java/jenkins/model/item_category/ItemCategory.java

```diff
@@ -0,0 +1,112 @@
+package jenkins.model.item_category;
+
+import hudson.Extension;
+import hudson.ExtensionPoint;
+import hudson.model.ModelObject;
+import jenkins.model.Messages;
+
+/**
+ * A category for {@link hudson.model.Item}s.
+ *
+ * @since TODO
+ */
+public abstract class ItemCategory implements ModelObject, ExtensionPoint {
+
+    public static int MIN_WEIGHT = 0;
+
+    public static int MIN_TOSHOW = 1;
+
+    /**
+     * Identifier, e.g. "category-id-standaloneprojects", etc.
+     *
+     * @return the identifier
+     */
+    public abstract String getId();
+
+    /**
+     * The description in plain text
+     *
+     * @return the description
+     */
+    public abstract String getDescription();
+
+    /**
+     * Helpful to set the order.
+     *
+     * @return the weight
+     */
+    public abstract int getWeight();
+
+    /**
+     * Minimum number required to show the category.
+     *
+     * @return the minimum items required
+     */
+    public abstract int getMinToShow();
+
+    /**
+     * The default {@link ItemCategory}, if an item doesn't belong anywhere else, this is where it goes by default.
+     */
+    @Extension
+    public static final class UncategorizedCategory extends ItemCategory {
+
+        @Override
+        public String getId() {
+            return "itemcategory-uncategorized";
+        }
+
+        @Override
+        public String getDescription() {
+            return Messages.ItemCategory_Uncategorized_Description();
+        }
+
+        @Override
+        public String getDisplayName() {
+            return Messages.ItemCategory_Uncategorized_DisplayName();
+        }
+
+        @Override
+        public int getWeight() {
+            return ItemCategory.MIN_WEIGHT;
+        }
+
+        @Override
+        public int getMinToShow() {
+            return ItemCategory.MIN_TOSHOW;
+        }
+
+    }
+
+    /**
+     * A generic {@link ItemCategory}
+     */
+    @Extension
+    public static final class StandaloneProjectCategory extends ItemCategory {
```

Comment:
:bug: please put this in a type that is specific to “projects”, such as `Job`, or as a top-level type, so that it could potentially be moved elsewhere in the future.